src/example/prompt.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class PullRequestReviewService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getPullRequestDetails(repo: string, prId: number): Promise<any> {
+    const url = `https://api.github.com/repos/${repo}/pulls/${prId}`;
+    return this.httpService.get(url).pipe(map(response => response.data)).toPromise();
```

Comment:
PullProofAI

Consider adding error handling for potential network issues or API failures when making HTTP requests.
```diff
+     return this.httpService.get(url).pipe(
+         map(response => response.data),
+         catchError(error => {
+             throw new Error('Failed to fetch pull request details');
+         })
+     ).toPromise();
```



<!-- This is an auto-generated comment by PullProofAI -->